contrib/mesos/pkg/executor/timer.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package executor
+
+import (
+	"time"
+)
+
+// optionalTimer is a convenience wrapper around time.Timer: it allows callers
+// to invoke funcs on nil references to *optionalTimer's without fear of a panic,
+// returning sane results upon invocations of those funcs.
```

Comment:
Why don't you replace the whole `duration==0` with a timer usind max-int64? That will run until eternity as well and is still a valid timer. Then we can completely get rid of this optionalTimer concept.